include/odp/api/spec/crypto.h

```diff
@@ -381,8 +394,15 @@ typedef struct odp_crypto_op_param_t {
 		/** Pointer to ADD */
 		uint8_t *ptr;
 
-		/** AAD length in bytes. Use odp_crypto_auth_capability() for
-		 *  supported AAD lengths. */
+		/** AAD length in bytes.
+		 *
+		 *  Used only when AAD length is variable within the session.
+		 *  When AAD length is constant, this must be set to zero and
+		 *  session level AAD length parameter is used instead.
+		 *
+		 *  Use odp_crypto_auth_capability() for supported AAD lengths.
+		 *  The default value is zero.
+		 */
 		uint32_t length;
```

Comment:
@psavol in dpdk 17.08 there is no support of variable aad length but only session specific aad length. ex: If application tries to set variable length instead of constant length, dpdk crypto operation will fail. Because once the crypto session is created with auth_aad_len=0, dpdk driver will use zero length in encryption_gcm & decryption_gcm.

Reply:
The spec says that if you are using a constant AAD length you must set that in the session AAD length, so the only reason why this still exists in the op parameters is to cover the (unlikely?) case of needing varying AAD lengths, which is a case that DPDK does not support.